### PR TITLE
feat: add flags to json schema

### PR DESCRIPTION
**Pull request by pquadri**
closes https://github.com/dbt-labs/dbt-jsonschema/issues/141

small note: i did not sort the keys with the extension as a good chunk of the file isn't and i didn't want to produce a huge diff

### CODE REVIEW

**Comment by aleix-cd** [comment]
Out of curiosity: shouldn't the `flags` object also accept all the global config flags specified [here](https://docs.getdbt.com/reference/global-configs/about-global-configs#available-flags) (where _Supported in project?_ = ✅ )?

**Comment by pquadri** [comment]
> Out of curiosity: shouldn't the `flags` object also accept all the global config flags specified [here](https://docs.getdbt.com/reference/global-configs/about-global-configs#available-flags) (where _Supported in project?_ = ✅ )?

guess so, i ignored the first part of https://github.com/dbt-labs/dbt-core/blob/d389ff1450e72c51eee443facf779736677768bb/core/dbt/contracts/project.py#L322 😅 , updating

**Comment by joellabes** [comment]
@pquadri thank you for opening this! (And sorry for the delay getting back to you). 

Looks great - just waiting for the checks to pass and I'll get this merged! There's some custom sort logic in [`.vscode/settings.json`](https://github.com/dbt-labs/dbt-jsonschema/blob/main/.vscode/settings.json#L110-L126) which if your sorter doesn't align with would lead to weird diffs. I ran it locally for you. 

I've also removed the flags from `dbt_project.json` which is legacy from <1.4 and only exists for backwards compatibility reasons.

File: schemas/latest/dbt_project-latest.json [hunk 1]
```diff
@@ -59,6 +59,80 @@
...
+          "type": "boolean"
+        },
+        "require_resource_names_without_spaces": {
+          "type": "boolean"
+        },
+        "send_anonymous_usage_stats": {
+          "type": "boolean"
+        },
+        "source_freshness_run_project_hooks": {
+          "type": "string"
```
Comment (joellabes) [inline schemas/latest/dbt_project-latest.json:108]:
```suggestion
          "type": "boolean"
```

**Review by joellabes — APPROVED** [review]

**Pull request merged**